scipy/optimize/_linprog_highs.py

```diff
@@ -335,21 +338,45 @@ def _linprog_highs(lp, solver, time_limit=None, presolve=True,
     else:
         slack, con = None, None
 
+    if 'lambda' in res:
+        # lagrange multipliers for equalities/inequalities
+        lamda = res['lambda']
+        ineqlin = np.array(lamda[:len(b_ub)])
+        eqlin = np.array(lamda[len(b_ub):])
+
+        # now get lagrange multipliers corresponding to bounds
+        upper = np.zeros(len(c))
+        lower = np.zeros(len(c))
+        bound_duals = res['s']
+        basis_statuses = res['basis_statuses']
+        for ii, (bnd, status) in enumerate(zip(bound_duals, basis_statuses)):
+            if status == HIGHS_BASIS_STATUS_LOWER:
+                lower[ii] = bnd
+            elif status == HIGHS_BASIS_STATUS_UPPER:
+                upper[ii] = bnd
+    else:
+        ineqlin, eqlin = None, None
+        upper, lower = None, None
+
     # this needs to be updated if we start choosing the solver intelligently
     solvers = {"ipm": "highs-ipm", "simplex": "highs-ds", None: "highs-ds"}
 
-    sol = {'x': np.array(res['x']) if 'x' in res else None,
-           'slack': slack,
-           # TODO: Add/test dual info like:
-           # 'lambda': res.get('lambda'),
-           # 's': res.get('s'),
```

Comment:
What was `s`?

Reply:
s is the same as in _highs_wrapper.pyx. It looks like once upon a time I had planned on passing it straight through, but this can now be removed

Reply:
More specifically, if `lambda` are the Lagrange multipliers associated with the constraints `Ax = b`, the `s` are the Lagrange multipliers associated with `x >= 0`

Reply:
OK. That information ends up in `lower`/`upper`?

Reply:
Sorry, I was incorrect before. s is returned be _highs_wrapper.pyx and _linprog_highs.py translates that into the lower/upper fields. So yes, that information makes it into lower/upper

Reply:
Now that I'm looking at this again, it might be something to ask Julian if I'm doing correctly. Each s value has a corresponding basis status value that tells whether it's lower or upper bound. There are other values, but I just set to 0 if it's not one of those two

Reply:
Hmm. Yeah good thing to check. Also ask what he thinks of the documentation and name `sensitivity`, please.